lib/app.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+module.exports = (host, verbose) => {
+    var dateService, logger, requestPromise, ShelfFile, shelfLib, ShelfMetadata, ShelfReference, URI, utils;
+
+    if (verbose) {
+        verbose = true;
+    } else {
+        verbose = false;
+    }
+
+    // If host doesn't start with "http://" or "https://", throw an error.
+    if (!/^http(|s):\/\//.test(host)) {
+        throw new Error("Shelf host must start with \"http://\" or \"https://\"");
+    }
+
+    shelfLib = {};
+    dateService = require("./date-service")();
+    logger = require("./logger")({
+        verbose
+    });
+    requestPromise = require("request-promise");
+    URI = require("urijs");
+    utils = require("./utils")(dateService, logger, URI);
+    ShelfMetadata = require("./shelf-metadata")(logger, requestPromise, utils);
+    ShelfFile = require("./shelf-file")(logger, requestPromise, ShelfMetadata, utils);
```

Comment:
Yeah, I suppose. I guess I was trying to be as explicit as possible about what they were. I'll remove the "shelf-" portion of stuffs.